### ISSUE TITLE
[BUG] reduce Node Status updates by truncating disk StorageAvailable

### ISSUE THREAD

**Issue by joshimoo** [longhorn/longhorn#2680]
TLDR: node status is constantly updated and never reaches a stable state.

This one is a bit of a behavior change, I picked 100MB arbitrarily could have used 10MB instead but it means it will report sooner / more often.

Since this is retrieved as part of the node_controller on block precision,
the node controller will be constantly updating the diskStatus since there
will always be something using disk space on the default disk (logs, etc)

Even for external disks, that don't have another consumer except longhorn
There is no need for block precision on StorageAvailable since it should not
be a problem to minimally under report the available storage.

This also addresses another cause of excessive websocket traffic longhorn/longhorn#2672

**Comment by joshimoo** [comment]
Test notes: observe that Node status available storage will be updated in 100MB increments (truncate to lower).
Example: 501mb available, will show as 500mb
Example: 599mb available, will show as 500mb

You should see this as part of the UI as well as API
This should also reduce the amount of websocket traffic further.

**Comment by joshimoo** [comment]
This one needs an E2E test update for the following tests:
# hasn't been updated, since the node available storage PR
test_node.py::test_node_disk_update FAILED
test_node.py::test_node_controller_sync_storage_available FAILED

tracked at https://github.com/longhorn/longhorn/issues/2692

**Comment by kaxing** [comment]
Manual validating with 1.1.2-head(longhorn-ui: 381c5aa)
1 control node + 3 worker nodes + 1 longhorn volume

By listening directly to the `/v1/ws/1s/nodes` while writing 20G data directly into /data of the testing pod,
here is the comparison from observation:

v1.1.1
Updating less than 1 second
After writing 20G continue to giving random updates

v1.1.2
The WebSocket connection updates every 15 seconds
No more update after writing test finished

A sample WebSocket connection log when writing 20GB to the volume:
(header: timestamp, `storageAvailable` on node1, node2, node3 in GiB)
```
"2021-06-17T07:44:30Z",41.9921875,42.3828125,41.6015625
"2021-06-17T07:44:45Z",41.9921875,41.69921875,40.234375
"2021-06-17T07:45:00Z",40.4296875,41.69921875,40.13671875
"2021-06-17T07:45:15Z",40.4296875,40.0390625,38.671875
"2021-06-17T07:45:30Z",38.8671875,40.0390625,38.671875
"2021-06-17T07:45:45Z",38.8671875,38.4765625,37.20703125
"2021-06-17T07:46:00Z",37.890625,38.4765625,37.20703125
"2021-06-17T07:46:03Z",37.79296875,37.40234375,37.20703125
"2021-06-17T07:46:15Z",37.79296875,37.40234375,36.1328125
"2021-06-17T07:46:30Z",36.328125,37.40234375,35.546875
"2021-06-17T07:46:33Z",36.23046875,35.9375,35.546875
"2021-06-17T07:46:45Z",36.23046875,35.9375,34.5703125
"2021-06-17T07:47:00Z",34.765625,35.9375,34.47265625
"2021-06-17T07:47:15Z",34.765625,34.375,33.0078125
"2021-06-17T07:47:30Z",33.30078125,34.375,33.0078125
"2021-06-17T07:47:45Z",33.30078125,32.91015625,31.54296875
"2021-06-17T07:48:00Z",31.8359375,32.91015625,31.54296875
"2021-06-17T07:48:15Z",31.73828125,31.4453125,31.54296875
"2021-06-17T07:48:16Z",31.73828125,31.4453125,29.98046875
"2021-06-17T07:48:31Z",30.2734375,30.56640625,29.98046875
"2021-06-17T07:48:45Z",30.2734375,29.8828125,28.61328125
"2021-06-17T07:49:00Z",28.80859375,29.8828125,28.515625
"2021-06-17T07:49:15Z",28.515625,28.41796875,27.1484375
"2021-06-17T07:49:30Z",27.34375,28.41796875,27.1484375
"2021-06-17T07:49:45Z",27.34375,27.05078125,25.68359375
"2021-06-17T07:50:00Z",25.87890625,27.05078125,25.68359375
"2021-06-17T07:50:03Z",25.78125,25.48828125,25.68359375
"2021-06-17T07:50:15Z",25.78125,25.48828125,24.21875
"2021-06-17T07:50:30Z",24.51171875,25.48828125,24.12109375
"2021-06-17T07:50:45Z",24.4140625,24.12109375,22.75390625
"2021-06-17T07:51:00Z",23.14453125,24.12109375,22.65625
"2021-06-17T07:51:15Z",23.14453125,22.94921875,22.16796875
```

**Comment by joshimoo** [comment]
@kaxing I don't know whether you did the rounding or not, to make it easier to read or whether this is the raw output.
But let me show just the calculation for the first and last line for each node:

```
node1:
41.9921875 * 1024 = 43000MB
23.14453125 * 1024 = 23700MB

node2:
42.3828125 * 1024 = 43300MB
22.94921875 * 1024 = 23500MB

node3:
41.6015625 * 1024 = 42600MB
22.16796875 * 1024 = 22700MB
```

From this we can see the delta is roughly 20GB (as you have written)
The small size difference can be based on when the websocket was started, as well as whether the volume was already created, i.e. has a filesystem on it already, which will consume initial space)

To the implementation, we round down the available space to the next 100MB:
So if the disk has 899MB available, it will be recorded as 800MB available.
So if the disk has 150MB available, it wil be recorded as 100MB available.

So the under reporting of available disk space is between [0,99]MB which for a storage system is nothing :)

**Comment by kaxing** [comment]
Validation - **Passed**

Testing against to:
> Manual validating with 1.1.2-head(longhorn-ui: 381c5aa)
> 1 control node + 3 worker nodes + 1 longhorn volume

Re-checking the last 5 raw data from the WS outputs:
| Time | available size change (mb) |
|:-:|:-:|
|07:50:30|-100|
|07:50:30|0|
|07:50:45|-1300|
|07:51:00|-100|
|07:51:15|-1300|




Thank you Joshua
